src/flick/lst/controllers/update_lst_controller.py

```diff
@@ -32,11 +33,14 @@ def _add_shows(self, show_ids):
                 show = Show.objects.get(pk=show_id)
                 self._lst.shows.add(show)
 
-    def _add_collaborators(self, collaborator_ids):
+    def _add_collaborators(self, collaborator_ids, request_user):
+        friends = [friend.id for friend in Friend.objects.friends(user=request_user)]
         self._lst.collaborators.clear()
         for c_id in collaborator_ids:
             if User.objects.filter(pk=c_id):
                 collaborator = User.objects.get(pk=c_id)
+                if c_id not in friends:
```

Comment:
i don't recommend getting all of the friends first then checking if the friends are in this list -- can you do a filter query instead on the Friend objects? and instead of checking not in friends, check `if Friend.objects.filter(user=...`?

Reply:
i'm not sure if the friend api we're using lets us filter though, so you should look into this!

Reply:
if this isn't possible though, i'd recommend changing `friends` to `friend_ids`